crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs

```diff
@@ -0,0 +1,114 @@
+use ruff_diagnostics::AlwaysFixableViolation;
+use ruff_diagnostics::{Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::whitespace::trailing_comment_start_offset;
+use ruff_python_ast::Stmt;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::fix;
+
+/// ## What it does
+/// Checks for unnecessary `pass` statements in functions, classes, and other
+/// blocks.
+///
+/// In [preview], this rule also checks for unnecessary ellipsis (`...`)
+/// expressions.
```

Comment:
Should we use the term "literal" here instead of "expression"? This applies to wherever this is being used.